packages/attestation-service/src/sms/twilio.ts

```diff
@@ -76,15 +119,76 @@ export class TwilioSmsProvider extends SmsProvider {
     } catch (error) {
       throw new Error(`Twilio Messaging Service could not be fetched: ${error}`)
     }
+    if (this.verifyServiceSid) {
+      try {
+        await this.client.verify.services
+          .get(this.verifyServiceSid)
+          .fetch()
+          .then((service) => {
+            if (!service.customCodeEnabled) {
+              // Make sure that custom code is enabled
+              throw new Error(
+                'TWILIO_VERIFY_SERVICE_SID is specified, but customCode is not enabled. Please contact Twilio support to enable it.'
+              )
+            }
+          })
+      } catch (error) {
+        throw new Error(`Twilio Verify Service could not be fetched: ${error}`)
+      }
+    }
   }
 
   async sendSms(attestation: AttestationModel) {
-    const m = await this.client.messages.create({
-      body: attestation.message,
-      to: attestation.phoneNumber,
-      from: this.messagingServiceSid,
-      statusCallback: this.deliveryStatusURL,
-    })
-    return m.sid
+    // Prefer Verify API if Verify Service is present and
+    // user is not requesting a deeplink
+    if (this.verifyServiceSid && attestation.securityCode) {
+      const requestParams: any = {
+        to: attestation.message,
+        channel: 'sms',
+        customCode: attestation.securityCode,
+      }
+
+      if (attestation.appSignature) {
+        requestParams.appHash = attestation.appSignature
```

Comment:
what does the verifyAPI do with this? is it just for logging/monitoring purposes?

Reply:
Twilio will add the <#> prefix and the apphash at the end if this is specified.
Example: `<#>Your Celo verification code is: 12345678 bU9E4ctGtIW`
(For Android devices)